public/Add-DbaReplArticle.ps1

```diff
@@ -0,0 +1,179 @@
+
+    .EXAMPLE
+        PS C:\> Add-DbaReplArticle -SqlInstance mssql1 -Database Northwind -Publication PubFromPosh -Name TableToRepl
+
+        Adds the TableToRepl table to the PubFromPosh publication from mssql1.Northwind
+
+
+    .EXAMPLE
+        PS C:\> $article = @{
+                    SqlInstance           = "mssql1"
+                    Database              = "pubs"
+                    Publication           = "testPub"
+                    Name                  = "publishers"
+                    Filter                = "city = 'seattle'"
+                }
+        PS C:\> Add-DbaReplArticle @article -EnableException
+
+        Adds the publishers table to the TestPub publication from mssql1.Pubs with a horizontal filter of only rows where city = 'seattle.
+
+    .EXAMPLE
+        PS C:\> $cso = New-DbaReplCreationScriptOptions -Options NonClusteredIndexes, Statistics
+        PS C:\> $article = @{
+                    SqlInstance           = 'mssql1'
+                    Database              = 'pubs'
+                    Publication           = 'testPub'
+                    Name                  = 'stores'
+                    CreationScriptOptions = $cso
+                }
+        PS C:\> Add-DbaReplArticle @article -EnableException
+
+        Adds the stores table to the testPub publication from mssql1.pubs with the NonClusteredIndexes and Statistics options set
+        includes default options.
+    #>
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess, ConfirmImpact = 'Medium')]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [parameter(Mandatory)]
+        [String]$Database,
+        [parameter(Mandatory)]
+        [String]$Publication,
+        [String]$Schema = 'dbo',
+        [parameter(Mandatory)]
+        [String]$Name,
+        [String]$Filter,
+        [PSObject]$CreationScriptOptions,
+        [Switch]$EnableException
+    )
+    process {
+
+        # Check that $CreationScriptOptions is a valid object
+        if ($CreationScriptOptions -and ($CreationScriptOptions -isnot [Microsoft.SqlServer.Replication.CreationScriptOptions])) {
+            Stop-Function -Message "CreationScriptOptions should be the right type. Use New-DbaReplCreationScriptOptions to create the object" -ErrorRecord $_ -Target $instance -Continue
+        }
+
+        foreach ($instance in $SqlInstance) {
+            try {
+                $replServer = Get-DbaReplServer -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+            Write-Message -Level Verbose -Message "Adding article $Name to publication $Publication on $instance"
+
+            try {
+                if ($PSCmdlet.ShouldProcess($instance, "Adding an article to $Publication")) {
```

Comment:
Having this be at the top of the workflow does not provide much detail on WhatIf output. Adding this if block to each logical check you are doing with data from the Get command call will provide more details to the user. This by nature will also improve the verbose output of the command too.

Reply:
Updated and this is the output - thought?

```PowerShell
Add-DbaReplArticle @article -whatif
What if: Performing the operation "Get the publication details for testpub" on target "mssql1".
What if: Performing the operation "Create an article object for testpub which is a Transactional publication" on target "mssql1".
What if: Performing the operation "Add filter for article: publishers" on target "mssql1".
What if: Performing the operation "Create article: publishers" on target "mssql1".
```